llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -2014,6 +2011,106 @@ MCCASBuilder::createDebugStringRefs() {
   return DebugStringContents;
 }
 
+static ArrayRef<char> getFragmentContents(const MCFragment &Fragment) {
+  switch (Fragment.getKind()) {
+#define MCFRAGMENT_NODE_REF(MCFragmentName, MCEnumName, MCEnumIdentifier)      \
+  case MCFragment::MCEnumName: {                                               \
+    const MCFragmentName &SF = cast<MCFragmentName>(Fragment);                 \
+    return SF.getContents();                                                   \
+  }
+#define MCFRAGMENT_ENCODED_FRAGMENT_ONLY
+#include "llvm/MCCAS/MCCASObjectV1.def"
+  case MCFragment::FT_CVInlineLines: {
+    const MCCVInlineLineTableFragment &SF =
+        cast<MCCVInlineLineTableFragment>(Fragment);
+    return SF.getContents();
+  }
+  case MCFragment::FT_LEB: {
+    const MCLEBFragment &SF = cast<MCLEBFragment>(Fragment);
+    return SF.getContents();
+  }
+  default:
+    return ArrayRef<char>();
+  }
+}
+
+static unsigned getRelocationSize(const MachO::any_relocation_info &RE,
+                                  bool IsLittleEndian) {
+  if (IsLittleEndian)
+    return 1 << ((RE.r_word1 >> 25) & 3);
+  return 1 << ((RE.r_word1 >> 5) & 3);
+}
+
+static uint32_t getRelocationOffset(const MachO::any_relocation_info &RE) {
+  return RE.r_word0;
+}
+
+/// This helper function is used to partition a Fragment into 2 parts, the \p
+/// FinalFragmentContents will contain the contents of the Fragment that will
+/// deduplicate and will be stored as part of a *FragmentRef or a
+/// MergedFragmentRef, the \p Addends are the location of all the relocation
+/// addends that will not deduplicate in the CAS and must be stored separately,
+/// there is one AddendRef per section in the CAS. The \p RelocationBufferIndex
```

Comment:
Maybe move this subsentence up front, and maybe include a little text art diagram?

Reply:
There are text art diagrams in MCCASObjectV1.h, I can add one there. I also have to update the diagram for the debug line section there. 

I am hoping @felipepiovezan can help update the diagram for the debug info section